src/Build/Definition/ProjectCollection.cs

```diff
@@ -176,6 +176,8 @@ public class ProjectCollection : IToolsetProvider, IBuildComponent, IDisposable
         /// </summary>
         private bool _onlyLogCriticalEvents;
 
+        private bool _enableTargetOutputLogging;
```

Comment:
Unfortunately, the logging service needing to be able to control the target output behavior means that every place that might conceivably be able to _initialize_ the logging service now needed to have that parameter.